Foundation/CharacterSet.swift

```diff
@@ -469,7 +469,7 @@ public struct CharacterSet : ReferenceConvertible, Equatable, Hashable, SetAlgeb
     
     /// Returns true if the two `CharacterSet`s are equal.
     public static func ==(lhs : CharacterSet, rhs: CharacterSet) -> Bool {
-        return lhs._wrapped.isEqual(rhs._bridgeToObjectiveC()) // TODO: mlehew - as  NSCharacterSet
+        return lhs._mapUnmanaged { l in rhs._mapUnmanaged { r in l.isEqual(r) } }
```

Comment:
I'll add a comment.

The problem was `rhs._bridgeToObjectiveC()` would always return an invalid (empty) CharacterSet.
`._mapUnmanaged` behaved correctly.

Now that I think about it, it can actually be changed to
```swift
return lhs._mapUnmanaged { $0.isEqual(rhs) }
```

PS
I actually fixed `._bridgeToObjectiveC()` in this PR too, so 
`lhs._wrapped.isEqual(rhs._bridgeToObjectiveC())` would start working correctly,
but I think `lhs._mapUnmanaged { $0.isEqual(rhs) }` follows better the pattern seen in other methods of `_SwiftNSCharacterSet`, like
```swift
    override func characterIsMember(_ member: unichar) -> Bool {
        return _mapUnmanaged { $0.characterIsMember(member) }
    }
    
    override func longCharacterIsMember(_ member: UInt32) -> Bool {
        return _mapUnmanaged { $0.longCharacterIsMember(member) }
    }
    
    override func isSuperset(of other: CharacterSet) -> Bool {
        return _mapUnmanaged { $0.isSuperset(of: other) }
    }
```

Does that make sense @itaiferber ?